autogpt_platform/backend/backend/blocks/video.py

```diff
@@ -0,0 +1,243 @@
+import os
+import tempfile
+from typing import Literal, Optional
+
+from moviepy.audio.io.AudioFileClip import AudioFileClip
+from moviepy.video.fx.Loop import Loop
+from moviepy.video.io.VideoFileClip import VideoFileClip
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import SchemaField
+from backend.util.file import get_path, store_temp_file
+
+
+class MediaDurationBlock(Block):
+
+    class Input(BlockSchema):
+        media_in: str = SchemaField(
+            description="Media input (URL, data URI, or local path)."
+        )
+        is_video: bool = SchemaField(
+            description="Whether the media is a video (True) or audio (False).",
+            default=True,
+        )
```

Comment:
Ideally we replace this with a bit of logic to determine the media type

Reply:
if you have the helper for this feel free to slap it here later